easybuild/framework/easyblock.py

```diff
@@ -2795,3 +2793,127 @@ class StopException(Exception):
     StopException class definition.
     """
     pass
+
+
+def inject_checksums(ecs, checksum_type):
+    """
+    Inject checksums of given type in specified easyconfig files
+
+    :param ecs: list of EasyConfig instances to inject checksums into corresponding files
+    :param checksum_type: type of checksum to use
+    """
+    for ec in ecs:
+        ec_fn = os.path.basename(ec['spec'])
+        print_msg("injecting %s checksums in %s" % (checksum_type, ec['spec']))
+
+        # get easyblock instance and make sure all sources/patches are available by running fetch_step
+        print_msg("fetching sources & patches for %s..." % ec_fn)
+        app = get_easyblock_instance(ec)
+        app.update_config_template_run_step()
+        app.fetch_step()
+
+        # check for any existing checksums, require --force to overwrite them
+        found_checksums = bool(app.cfg['checksums'])
+        for ext in app.exts:
+            found_checksums |= bool(ext.get('checksums'))
+        if found_checksums:
+            if build_option('force'):
+                print_warning("Found existing checksums in %s, overwriting them (due to use of --force)..." % ec_fn)
+            else:
+                raise EasyBuildError("Found existing checksums, use --force to overwrite them")
+
+        # back up easyconfig file before injecting checksums
+        ectxt = read_file(ec['spec'])
+        ec_backup = os.path.join(ec['spec'] + '.bck')
+        if os.path.exists(ec_backup):
+            raise EasyBuildError("Backup of easyconfig already exists, please (re)move it first: %s", ec_backup)
+        else:
+            write_file(ec_backup, ectxt)
+        print_msg("backup of easyconfig file saved to %s..." % ec_backup)
+
+        # compute & inject checksums for sources/patches
+        print_msg("computing & injecting %s checksums for sources & patches for %s..." % (checksum_type, ec_fn))
+        checksums = []
+        for entry in app.src + app.patches:
+            checksum = compute_checksum(entry['path'], checksum_type)
+            print_msg("* %s: %s" % (os.path.basename(entry['path']), checksum))
+            checksums.append((os.path.basename(entry['path']), checksum))
+
+        if len(checksums) == 1:
+            checksum_lines = ["checksums = ['%s']\n" % checksums[0][1]]
+        else:
+            checksum_lines = ['checksums = [']
+            for fn, checksum in checksums:
+                checksum_lines.append("    '%s',  # %s" % (checksum, fn))
+            checksum_lines.append(']\n')
+
+        checksums_txt = '\n'.join(checksum_lines)
+
+        if app.cfg['checksums']:
+            regex = re.compile(r'^checksums(?:.|\n)*?\]\s*$', re.M)
+            ectxt = regex.sub(checksums_txt, ectxt)
+        elif app.src:
```

Comment:
no, it's possible there are no sources, and hence no patches/checksums

for example: bundles, like `intel-2017a.eb` easyconfig